insta/src/macros.rs

```diff
@@ -74,14 +74,14 @@ macro_rules! assert_toml_snapshot {
     };
 }
 
-/// Asserts a `Serialize` snapshot in YAML format.
+/// Asserts a [`serde::Serialize`] snapshot in YAML format.
 ///
 /// **Feature:** `yaml`
 ///
-/// The value needs to implement the `serde::Serialize` trait and the snapshot
+/// The value needs to implement the [`serde::Serialize`] trait and the snapshot
 /// will be serialized in YAML format.  This does mean that unlike the debug
 /// snapshot variant the type of the value does not appear in the output.
-/// You can however use the `assert_ron_snapshot!` macro to dump out
+/// You can however use the [`crate::assert_ron_snapshot!`] macro to dump out
```

Comment:
All these `crate` specs are too much structure iff the links work without them. Fine if they're required on balance

Reply:
Jea.. I have no clue why they are required..
(the other references are equivalently nessesary sadly.. maybe marcros and scoping? Would need to debug further, but currently don't have time for diving into rustdoc^^)

This seems more like a bug, right?

![image](https://github.com/user-attachments/assets/a22b264c-b2f5-4819-960e-85e97950882c)

Reply:
Is the feature enabled?

Reply:
Yes, have just double checked myself. I did activate all features:

```bash
cargo doc --all-features --all --keep-going 
```

Reply:
This is also reproducible via `make doc`, aka
```bash
RUSTC_BOOTSTRAP=1 RUSTDOCFLAGS="--cfg=docsrs" cargo doc --no-deps --all-features
```

Reply:
Weirdly I can't repro this...

Reply:
Here are details to reproduce :

<details><summary><code>rustup --version<code><summary>

```
rustup 1.27.1 (54dd3d00f 2024-04-24)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.81.0 (eeb90cda1 2024-09-04)`
```

<details>

<details><summary><code>git diff | cat<code><summary>

```
diff --git a/insta/src/macros.rs b/insta/src/macros.rs
index 45434b9..9c69005 100644
--- a/insta/src/macros.rs
+++ b/insta/src/macros.rs
@@ -81,7 +81,7 @@ macro_rules! assert_toml_snapshot {
 /// The value needs to implement the [`serde::Serialize`] trait and the snapshot
 /// will be serialized in YAML format.  This does mean that unlike the debug
 /// snapshot variant the type of the value does not appear in the output.
-/// You can however use the [`assert_ron_snapshot!`](crate::assert_ron_snapshot!) macro to dump out
+/// You can however use the [`assert_ron_snapshot!`] macro to dump out
 /// the value in [RON](https://github.com/ron-rs/ron/) format which retains some
 /// type information for more accurate comparisons.
 ///
```

<details>

<details><summary><code>RUSTC_BOOTSTRAP=1 RUSTDOCFLAGS="--cfg=docsrs" cargo doc --no-deps --all-features --keep-going<code><summary>

```
    Checking insta v1.41.0 (/tmp/insta/insta)
 Documenting insta v1.41.0 (/tmp/insta/insta)
 Documenting cargo-insta v1.41.0 (/tmp/insta/cargo-insta)
warning: unresolved link to `assert_ron_snapshot`
   --> insta/src/macros.rs:77:1
    |
77  | / /// Asserts a [`serde::Serialize`] snapshot in YAML format.
78  | | ///
79  | | /// **Feature:** `yaml`
80  | | ///
...   |
120 | | ///
121 | | /// The snapshot name is optional but can be provided as first argument.
    | |________________________________________________________________________^
    |
    = note: the link appears in this line:
            
            You can however use the [`assert_ron_snapshot!`] macro to dump out
                                     ^^^^^^^^^^^^^^^^^^^^^^
    = note: no item named `assert_ron_snapshot` in scope
    = note: `macro_rules` named `assert_ron_snapshot` exists in this crate, but it is not in scope at this link's location
note: the lint level is defined here
   --> insta/src/lib.rs:2:9
    |
2   | #![warn(rustdoc::all)]
    |         ^^^^^^^^^^^^
    = note: `#[warn(rustdoc::broken_intra_doc_links)]` implied by `#[warn(rustdoc::all)]`

warning: `insta` (lib doc) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.19s
   Generated /tmp/insta/target/doc/cargo_insta/index.html and 1 other file
```

<details>

Reply:
Ah, sorry, I hadn't realized this was with changed code